src/dodal/beamlines/p99.py

```diff
@@ -32,3 +40,30 @@ def sample_stage() -> XYZPositioner:
 @device_factory()
 def lab_stage() -> XYZPositioner:
     return XYZPositioner(f"{PREFIX.beamline_prefix}-MO-STAGE-02:LAB:")
+
+
+andor_data_path = StaticPathProvider(
+    filename_provider=AutoIncrementFilenameProvider(base_filename="andor2"),
+    directory_path=Path("/dls/p99/data/2024/cm37284-2/processing/writenData"),
+)
+
+
+@device_factory()
+def andor2_det() -> Andor2Detector:
+    """Andor model:DU897_BV."""
+    return Andor2Detector(
+        prefix=f"{PREFIX.beamline_prefix}-EA-DET-03:",
+        path_provider=andor_data_path,
+        drv_suffix="CAM:",
+        fileio_suffix="HDF5:",
+    )
```

Comment:
```suggestion
set_path_provider(
    StaticVisitPathProvider(
        BL,
        Path("/dls/p99/data/2024/cm37284-2/processing/writenData"),
        client=RemoteDirectoryServiceClient("http://p99-control:8088/api"),
    )
)



@device_factory()
def andor2_det() -> Andor2Detector:
    """Andor model:DU897_BV."""
    return Andor2Detector(
        prefix=f"{PREFIX.beamline_prefix}-EA-DET-03:",
        path_provider=get_path_provider(),
        drv_suffix="CAM:",
        fileio_suffix="HDF5:",
    )
```

For consistency with other beamlines, and to enable blueapi to inject a PathProvider before instantiating the module (after discussions about this in #991 are resolved). Does p99 GDA build and is it run on the control machine at the moment?

Reply:
p99 does have gda running on the control machine, am I right that with this change, if I take multiple images within a single scan they would all be given the same path/filename?

Reply:
Actually in this case yes..... I was thinking about tiff format that use on i10...

Reply:
actually actually, I get this error `ClientConnectorError: Cannot connect to host p99-control:8088 ssl:default [Connect call failed ('172.23.177.32', 8088)]` So I am not sure what's my option anymore.

Reply:
https://github.com/DiamondLightSource/p99-bluesky/issues/79#issuecomment-2742966421